packages/salesforcedx-vscode-core/package.json

```diff
@@ -215,7 +215,7 @@
         },
         {
           "command": "sfdx.force.apex.class.create",
-          "when": "explorerResourceIsFolder && resourceFilename == classes && sfdx:project_opened"
+          "when": "explorerResourceIsFolder && resourcePath =~ /classes/ && sfdx:project_opened"
```

Comment:
This could also be further protected by adding more restrictions to the regex.  We could look for the presence of a directory ( windows or linux ) before the word `classes`.  

```resource =~ /(\/|\\\\)classes/```

But I just wasn't sure if that was overly protective when this regex reads easier.

Reply:
One thing I found was that if I created a `classes` directory in aura (so `/app/main/default/aura/classes`) the "SFDX: Create Apex Class" command is available.  Though not appropriate, I don't expect users will be creating folders called "classes".  (just something to be aware of).  I guess let's add it in and then see if we get any issues logged.

Reply:
It could also get fired if someone created a directory called 'myclasses' since there is no check for directory separators.  But I too was wondering what is over-protective compared to actual user behavior.